pkg/ottl/ottlfuncs/func_scale.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type ScaleArguments[K any] struct {
+	Value      ottl.Getter[K]
+	Multiplier float64
+}
```

Comment:
I think if we make this an editor, an optional `unit` could be good here.

Reply:
small question regarding this: This requires a modification of both the `data_points` and the `unit` property of a metric. Is there a way to access the complete metric via an OTTL statement? In the [docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/contexts/ottlmetric) I only found a way to access either the `data_points` or the `unit`:

```
processors:
  transform:
    error_mode: ignore
    metric_statements:
      - context: metric
        statements:
          - scale_metric(data_points, 10.0) # this does not give access to the unit of the metric
```

I guess a workaround would be to set the unit in a separate function, but if somehow possible I would like to also add the possibility to set the unit here

Reply:
We'd have access to the metric in the TransformContext since this is a metrics-only function. So we could allow something like `scale_metric(10.0, unit="s")`.  

It can always be added later, so we could drop this idea until someone asks for it.